contentcuration/contentcuration/tests/test_rectify_source_field_migraiton_command.py

```diff
@@ -0,0 +1,186 @@
+            "original_channel_id",
+            "original_source_node_id",
+            "coalesced_provider",
+            "coalesced_author",
+            "coalesced_aggregator",
+            "coalesced_license_id",
+            "original_source_node_f_changed",
+        ).order_by()
+
+        for item in final_nodes:
+            base_node = ContentNode.objects.get(pk=item["id"])
+            original_source_channel_id = item["original_channel_id"]
+            original_source_node_id = item["original_source_node_id"]
+            tree_id = (
+                Channel.objects.filter(pk=original_source_channel_id)
+                .values_list("main_tree__tree_id", flat=True)
+                .get()
+            )
+            original_source_node = ContentNode.objects.filter(
+                tree_id=tree_id, node_id=original_source_node_id
+            )
+
+            base_channel = Channel.objects.get(pk=item['channel_id'])
+
+            to_be_republished = not (base_channel.main_tree.get_family().filter(changed=True).exists())
+            print("onga bonga 2")
+            print(base_channel.main_tree.get_family().filter(changed=True))
+            print(to_be_republished)
+            if original_source_channel_id is not None and original_source_node.exists():
+                # original source node exists and its source fields dont match
+                # update the base node
+                if base_node.author != original_source_node[0].author:
+                    base_node.author = original_source_node[0].author
+                if base_node.provider != original_source_node[0].provider:
+                    base_node.provider = original_source_node[0].provider
+                if base_node.aggregator != original_source_node[0].aggregator:
+                    base_node.aggregator = original_source_node[0].aggregator
+                if base_node.license != original_source_node[0].license:
+                    base_node.license = original_source_node[0].license
+
+                base_node.save()
+
+                if to_be_republished and base_channel.published:
+                    # we would repbulish the channel
+                    print("publishingg the channel!!")
+                    publish_channel(self.user.id, base_channel.id)
+            else:
+                continue
+
+    def test_two_node_case(self):
+        base_channel = testdata.channel()
+        license_changed = License.objects.all()[1]
+        base_node = ContentNode.objects.create(
+            id=uuid.uuid4().hex,
+            title="base contentnode",
+            parent=base_channel.main_tree,
+            kind_id=content_kinds.VIDEO,
+            license=license_changed,
+            original_channel_id=self.original_channel.id,
+            source_channel_id=self.original_channel.id,
+            source_node_id=self.original_contentnode.node_id,
+            original_source_node_id=self.original_contentnode.node_id,
+        )
```

Comment:
One suggestion would be to use the same copy/import utilities that we use elsewhere, then override the things that shouldn't have changed, but it isn't a big deal. From my perspective, I like to do my best to ensure the tests are founded upon the app's behaviors as much as possible, because too many differences could cause the tests to pass when they shouldn't (under the typical behaviors of the app)